### PR TITLE
mgr/dashboard: Use htmllint

### PR DESCRIPTION
**Pull request by tspmelo**
This package will replace both html-linter and htmlhint.

Some rules from the previous packages were not working that well.
This new package has more rules and seems to work better.

Signed-off-by: Tiago Melo <tmelo@suse.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`

</details>

### CODE REVIEW

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/block/rbd-configuration-form/rbd-configuration-form.component.html [hunk 1]
```diff
@@ -6,7 +6,8 @@
        class="col-12">
     <h3 class="cd-header">
       <span (click)="toggleSectionVisibility(section.class)"
-        class="collapsible">{{ section.heading }} <i [ngClass]="!sectionVisibility[section.class] ? icons.addCircle : icons.minusCircle" aria-hidden="true"></i></span>
+        class="collapsible">{{ section.heading }} <i [ngClass]="!sectionVisibility[section.class] ? icons.addCircle : icons.minusCircle"
```
Comment (s0nea) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/block/rbd-configuration-form/rbd-configuration-form.component.html]: Is the indentation correct here?

Comment (tspmelo) [reply src/pybind/mgr/dashboard/frontend/src/app/ceph/block/rbd-configuration-form/rbd-configuration-form.component.html]: Not really, fixed it.
Went back to test the other plugins and seems none of them do a good job validating the indentation of attributes.

Comment (s0nea) [reply src/pybind/mgr/dashboard/frontend/src/app/ceph/block/rbd-configuration-form/rbd-configuration-form.component.html]: > Went back to test the other plugins and seems none of them do a good job validating the indentation of attributes.

Meh :/ But thank you!

**Comment by tspmelo** [comment]
jenkins test dashboard

**Comment by LenzGr** [comment]
Thank you! It looks as if your PR does make changes to both the test configuration as well as fixing all reported issues in a single commit. Would it make sense to separate this into two commits, one that changes the linter configuration and one that addresses all things that the new linter reported?

**Comment by LenzGr** [comment]
jenkins test dashboard

**Comment by tspmelo** [comment]
I prefer to have them in the same PR, since separating them would make one of the commits to fail if it was tested individually.

**Comment by tspmelo** [comment]
jenkins retest this please

**Comment by votdev** [comment]
jenkins retest this please

File: src/pybind/mgr/dashboard/frontend/.htmllintrc [hunk 2]
```diff
@@ -0,0 +1,70 @@
+{
+    "plugins": [],  // npm modules to load
+
```
Comment (votdev) [inline src/pybind/mgr/dashboard/frontend/.htmllintrc]: Remove this empty line in a follow up PR.

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/osd/osd-performance-histogram/osd-performance-histogram.component.html [hunk 3]
```diff
@@ -1,8 +1,6 @@
 <table>
-  <tr style="height: 10px;"
-      *ngFor="let row of valuesStyle">
-    <td style="width: 10px; height: 10px;"
-        *ngFor="let col of row"
+  <tr *ngFor="let row of valuesStyle">
```
Comment (votdev) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/osd/osd-performance-histogram/osd-performance-histogram.component.html]: Relocating styling into CSS :+1:

**Pull request merged**